projects/batfish-common-protocol/src/main/java/org/batfish/common/Container.java

```diff
@@ -5,36 +5,35 @@
 import com.google.common.base.MoreObjects;
 import java.nio.file.Path;
 import java.util.Objects;
-import java.util.SortedSet;
 
 public final class Container {
   private static final String NAME_VAR = "name";
-  private static final String TESTRIGS_VAR = "testrigs";
+  private static final String TESTRIGS_URI_VAR = "testrigsUri";
   private static final String ANALYSIS_VAR = "analysis";
 
   private String _name;
-  private SortedSet<String> _testrigs;
+  private String _testrigsUri;
   private Path _analysis;
 
   @JsonCreator
-  public static Container of(
-      @JsonProperty(NAME_VAR) String name, @JsonProperty(TESTRIGS_VAR) SortedSet<String> testrigs) {
-    return new Container(name, testrigs);
+  public static Container makeContainer(
+      @JsonProperty(NAME_VAR) String name, @JsonProperty(TESTRIGS_URI_VAR) String testrigsUrl) {
```

Comment:
This is my main concern, if a container object contains a list of Testrig Objects, the nested structure may make a container object really large. (e.g Testrig object contains a list of questions, a list of configs, etc.) We might only need some basic information of a container in some function calls (Testrig names in it for example). Have such a structure will cause some unnecessary informations in the network traffic. However, have a URI representation of inner resource also have some problems. For example, list all file names in a testrig requires additional calls to our web service (one for configs, one for questions, etc.) Also, there are some other issues for building a RESTful API for our existing service, I'll save them for later.